### PR TITLE
chore(dashboard): use step slug in the url

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
Dashboard: use the step slug in the URL.

### Screenshots

https://github.com/user-attachments/assets/137eb6aa-587f-4fe8-94ba-6c1abc9071dc

### CODE REVIEW

File: apps/dashboard/src/api/workflows.ts [hunk 1]
```diff
@@ -55,14 +55,14 @@ export const updateWorkflow = async ({
 export const previewStep = async ({
   workflowSlug,
   payload,
-  stepId,
+  stepSlug,
 }: {
   workflowSlug: string;
-  stepId: string;
+  stepSlug: string;
```
Comment (LetItRock) [inline apps/dashboard/src/api/workflows.ts:62]: the preview endpoint supports passing the stepSlug 🎉

File: apps/dashboard/src/components/workflow-editor/schema.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ export const workflowSchema = z.object({
         type: z.nativeEnum(StepTypeEnum),
         _id: z.string(),
         stepId: z.string(),
+        slug: z.literal<StepResponseDto['slug']>('_stp_'),
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/schema.ts:33]: the `slug` is used in the workflow nodes, we pass it to be able to build the navigation links, see the code in a file above

**Review by ChmaraX — APPROVED** [review]

File: apps/dashboard/src/components/workflow-editor/schema.ts (hunk 2 above)
Comment (LetItRock) [reply apps/dashboard/src/components/workflow-editor/schema.ts:33]: the `StepResponseDto["slug"]` is a template literal which requires this kind of template

File: apps/dashboard/src/context/environment/environment-provider.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { useCallback, useLayoutEffect, useMemo, useState } from 'react';
 import { useLocation, useNavigate, useParams } from 'react-router-dom';
-
-import { EnvironmentEnum, type IEnvironment } from '@novu/shared';
```
Comment (LetItRock) [inline apps/dashboard/src/context/environment/environment-provider.tsx:4]: `EnvironmentEnum` is failing the app and build, not sure why

**Review by SokratisVidros — APPROVED** [review]
🐌

**Pull request merged**